src/arch/mod.rs

```diff
@@ -31,24 +31,55 @@ lazy_static! {
 mod x64;
 pub use x64::*;
 
-pub fn union_size(symbols: &[Symbol]) -> Result<SIZE_T, &'static str> {
-    symbols
-        .iter()
-        .map(|symbol| symbol.ctype.sizeof())
-        // max of member sizes
-        .try_fold(1, |n, size| Ok(max(n, size?)))
-}
-
-pub fn struct_size(symbols: &[Symbol]) -> Result<SIZE_T, &'static str> {
-    symbols.iter().try_fold(0, |offset, symbol| {
-        Ok(Type::next_offset(offset, &symbol.ctype)?)
-    })
-}
+impl StructType {
+    /// Get the offset of the given struct member.
+    pub fn struct_offset(&self, member: super::InternedStr) -> u64 {
+        let members = self.members();
+        let mut current_offset = 0;
+        for formal in members.iter() {
+            if formal.id == member {
+                return current_offset;
+            }
+            current_offset = Self::next_offset(current_offset, &formal.ctype)
+                .expect("structs should have valid size and alignment");
+        }
+        unreachable!("cannot call struct_offset for member not in struct");
+    }
+    /// Get the offset of the next truct member given the current offset.
+    fn next_offset(mut current_offset: u64, ctype: &super::Type) -> Result<u64, &'static str> {
```

Comment:
Why did this change to `super::Type`?

Reply:
Its a remnant from `src/data/types.rs`

Reply:
Can you change it back? Ditto for `super::InternedStr`.